agave_app/agaveGui.cpp

```diff
@@ -803,8 +802,19 @@ agaveGui::openRecentFile()
       // assume that .obj is mesh
       openMesh(path);
     } else {
-      // assumption of ome.tif
-      if (!open(path.toStdString())) {
+      LoadingDialogType type;
+      QString extension = path.sliced(path.lastIndexOf("."));
+      if (path.indexOf("http") == 0) {
+        type = LoadingDialogType::URL;
+      } else if (extension.indexOf(".zarr") == 0) {
+        type = LoadingDialogType::DIRECTORY;
+      } else if (extension.indexOf(".json") == 0) {
+        type = LoadingDialogType::JSON;
+      } else {
+        type = LoadingDialogType::FILE;
+      }

```

Comment:
nit: this could be a function getLoadDialogTypeFromFilePath

Reply:
... could the load dialog figure this out for itself?

Reply:
You're right, it could. I'll move the code logic for it.